velox/functions/prestosql/aggregates/tests/PrestoHasherTest.cpp

```diff
@@ -0,0 +1,344 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/functions/prestosql/aggregates/PrestoHasher.h"
+#include "velox/functions/prestosql/aggregates/tests/AggregationTestBase.h"
+
+namespace facebook::velox::aggregate::test {
+template <typename T>
+using limits = std::numeric_limits<T>;
+
+class PrestoHasherTest : public AggregationTestBase {
+ protected:
+  template <typename T>
+  void assertHash(
+      const std::vector<std::optional<T>>& data,
+      const std::vector<int64_t>& expected) {
+    auto vector = makeNullableFlatVector<T>(data);
+    assertHash(vector, expected);
+  }
+
+  void assertHash(
+      const VectorPtr& vector,
+      const std::vector<int64_t>& expected) {
+    auto vectorSize = vector->size();
+
+    auto checkHashes = [&](const VectorPtr& vector,
+                           const std::vector<int64_t>& expected) {
+      auto vectorSize = vector->size();
+      SelectivityVector rows(vectorSize);
+      PrestoHasher hasher(*vector, rows);
```

Comment:
Do we have tests that use a subset of rows? It looks like all tests always specify all rows.